core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1515,8 +1515,10 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
   def getBucketUsage(workspaceName: WorkspaceName): Future[PerRequestMessage] = {
     for {
       bucketName <- dataSource.inTransaction { dataAccess =>
-        withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Write, dataAccess) { workspaceContext =>
-          DBIO.successful(workspaceContext.workspace.bucketName)
+        withWorkspaceContext(workspaceName, dataAccess) { workspaceContext =>
+          requireAccessIgnoreLock(workspaceContext.workspace, WorkspaceAccessLevels.Write, dataAccess) {
```

Comment:
We have to be careful to revisit this if another role is added. Specifically, the storage cost estimate story says that it should be displayed to anyone with more than reader access.